src/session_impl.cpp

```diff
@@ -1685,8 +1691,13 @@ namespace aux {
 					!= m_settings.get_str(settings_pack::listen_interfaces));
 
 		apply_pack(&pack, m_settings, this);
+
 		m_disk_thread.set_settings(&pack, m_alerts);
 
+#if TORRENT_USE_INVARIANT_CHECKS
+		check_invariant();
```

Comment:
I don't know which settings impl methods are available at the interface. I did put the checks to the deepest method call.

if apply_settings_pack is not visible too, then I would put it outside of it.